cli/setup.go

```diff
@@ -27,14 +27,23 @@ var setupCommand = cli.Command{
 			Name:  "clean",
 			Usage: "Removes bitrise's workdir before setup.",
 		},
+		cli.BoolFlag{
+			Name:  "minimal",
+			Usage: "Only installs the required tools for running in CI mode.",
+		},
 	},
 }
 
 func setup(c *cli.Context) error {
-	fullMode := c.Bool("full")
-	cleanMode := c.Bool("clean")
+	clean := c.Bool("clean")
+	minimal := c.Bool("minimal")
+
+	var setupMode = bitrise.SetupModeDefault
+	if minimal {
+		setupMode = bitrise.SetupModeMinimal
+	}
 
-	if err := bitrise.RunSetup(c.App.Version, fullMode, cleanMode); err != nil {
+	if err := bitrise.RunSetup(c.App.Version, setupMode, clean); err != nil {
```

Comment:
My idea here is that there are setup modes (an enum), and there is the clean flag, which is an addition to the setup mode and compatible with all values.

Reply:
I agree with this, but if we take that the setup mode is an enum (an not a minimal - not minimal boolean), then this log won't scale well when we add more setup modes in the future: https://github.com/bitrise-io/bitrise/pull/944/files#diff-f5bdaface1e2f3d8215896e7b459014d12afe5a7679dafd2b2bd84702b0baf4cR56